core/kits/documents/kit.php

```diff
@@ -62,6 +62,12 @@ public function get_css_wrapper_selector() {
 	}
 
 	public function save( $data ) {
+		foreach ( $this->tabs as $tab ) {
+			if ( method_exists( $tab, 'before_save' ) ) {
```

Comment:
Instead of that, create a protected empty `before_save` method inside `Tab_Base`

Reply:
It can't be void, it has to return a value.

Reply:
Done. Implemented a generic function.